### PR TITLE
Fix an NPE in ShadowConnectivityManager

### PR DESCRIPTION
**Pull request by hoisie**
If ConnectivityManager.getActiveNetwork() was called after calling ShadowConnectivityManager.setActiveNetworkInfo(null), an NPE would result. Fix this edge case.

### CODE REVIEW

File: robolectric/src/test/java/org/robolectric/shadows/ShadowConnectivityManagerTest.java [hunk 1]
```diff
@@ -839,4 +839,10 @@ public void defaults_afterReset() {
       assertThat(connectivityManager.getActiveNetwork()).isNotNull();
     }
   }
+
+  @Test
+  public void getActiveNetwork_afterSetActiveNetworkInfoNull() {
+    shadowOf(connectivityManager).setActiveNetworkInfo(null);
+    assertThat(connectivityManager.getActiveNetwork()).isNull();;
```
Comment (MGaetan89) [inline robolectric/src/test/java/org/robolectric/shadows/ShadowConnectivityManagerTest.java:846]:
```suggestion
    assertThat(connectivityManager.getActiveNetwork()).isNull();
```

**Review by MGaetan89 — APPROVED** [review]
I think the failing tests are flaky tests

File: robolectric/src/test/java/org/robolectric/shadows/ShadowConnectivityManagerTest.java (hunk 1 above)
Comment (hoisie) [reply robolectric/src/test/java/org/robolectric/shadows/ShadowConnectivityManagerTest.java]: Nice catch, updated this.

**Review by MGaetan89 — APPROVED** [review]
